package ConnSvr_Conf;

enum MsgCmd
{
	connsvr_start_req = 0;
	connsvr_start_rsp = 1;
	connsvr_stop	  = 2;
	connsvr_route	  = 3;
	connsvr_msg_ntf   = 4;
	connsvr_cmd_count = 5;
}

message StartReq
{
	required uint32 currentconn = 1;//当前已建立连接总数，不包括这个正在建立的连接
	required uint32 channelcnt = 2;//当前通道总数编号总是从0-cnt 0 总是默认
}

message StartRsp
{
	required int32 accept			= 1;//0接受，1拒绝
	required uint32 routechannel	= 2;//路由通道
}

message Stop
{
	required uint64 timestamp = 1;//发送时的时间
}

message Route
{
	required uint32 routechannel = 1;
}
message MsgNtf
{
	required bytes buff = 1;
}

message MsgHead
{
	required fixed32 cmdid  = 2;
	required fixed32 connid = 3;
	required uint32  port	= 4;
	required string  ip		= 5;
}

message ConnsvrMsg
{
	required MsgHead head		= 1;
	optional StartReq startreq	= 2;
	optional StartRsp startrsp	= 3;
	optional Stop stop			= 4;
	optional Route route		= 5;
	optional MsgNtf msgntf		= 6;
}

